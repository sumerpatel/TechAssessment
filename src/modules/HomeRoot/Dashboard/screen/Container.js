import React from 'react';
import { connect } from 'react-redux';
import Dashboard from './Dashboard';
import { actions } from '../action';
import { getDetails } from '../selectors';

export class DashboardContainer extends React.Component {
  render() {
    return <Dashboard {...this.props} />;
  }
}

const mapStateToProps = (state) => {
  return {
    userData: getDetails(state),
  };
};

export const mapDispatchToProps = (dispatch) => ({
  getUserData: () => dispatch(actions.getUserData()),
});

export default connect(mapStateToProps, mapDispatchToProps)(DashboardContainer);
