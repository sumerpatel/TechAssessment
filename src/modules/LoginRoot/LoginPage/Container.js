import React from 'react';
import { connect } from 'react-redux';
import Login from './Login';

export class LoginContainer extends React.Component {
  render() {
    return <Login {...this.props} />;
  }
}

const mapStateToProps = (state) => {
  return {};
};

export const mapDispatchToProps = (dispatch) => ({});

export default connect(mapStateToProps, mapDispatchToProps)(LoginContainer);
