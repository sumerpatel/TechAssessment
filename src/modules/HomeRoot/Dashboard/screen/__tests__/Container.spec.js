import React from 'react';
import { shallow } from 'enzyme';
import Container from '../Container';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();
const initialState = {};
describe('Container', () => {
  let wrapper, store;
  const props = {
    userData: [],
    getUserData: jest.fn()
  };
  beforeEach(() => {
    store = mockStore(initialState);
    wrapper = shallow(<Container store={store} {...props} />);
  });

  it('should render', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
