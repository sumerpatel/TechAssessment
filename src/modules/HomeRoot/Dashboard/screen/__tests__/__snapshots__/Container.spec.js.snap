// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should render 1`] = `
<ContextProvider
  value={null}
>
  <DashboardContainer
    getUserData={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
